### PR TITLE
Add OpenCL context

### PR DESCRIPTION
**Pull request by manhnt9**
Add OpenCL context creation to function `getCLDeviceFromGLContext` because:
- Reuse cl_context_properties and cl_device_id

Thus it's not only `getCLDeviceFromGLContext` anymore => function name changed to `initOpenCL`.

### CODE REVIEW

File: source/Irrlicht/COpenCLHandler.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ clGetGLContextInfoKHR_fn COpenCLHandler::pClGetGLContextInfoKHR = NULL;
 cl_platform_id COpenCLHandler::platforms[IRR_MAX_OCL_PLATFORMS];
 cl_uint COpenCLHandler::actualPlatformCount = 0;
 COpenCLHandler::SOpenCLPlatformInfo COpenCLHandler::platformInformation[IRR_MAX_OCL_PLATFORMS];
+cl_context COpenCLHandler::openCLContext = 0;
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/COpenCLHandler.cpp:17]: this shoud not be a static/singleton multiple contexts should be possible

File: source/Irrlicht/COpenCLHandler.h [hunk 2]
```diff
@@ -211,8 +211,10 @@ class COpenCLHandler
 
         static const SOpenCLPlatformInfo& getPlatformInfo(const size_t& ix) {return platformInformation[ix];}
 
+        static cl_context getOpenCLContext() {return openCLContext;}
+
 #ifdef _IRR_COMPILE_WITH_OPENGL_
-        static bool getCLDeviceFromGLContext(cl_device_id& outDevice,
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/COpenCLHandler.h:215]: we need both functions, for example when the OpenCL device has already been created externally and we need to see if its the right one for sharing with the OpenGL GPU.

File: source/Irrlicht/COpenGLDriver.cpp [hunk 3]
```diff
@@ -445,7 +445,7 @@ bool COpenGLDriver::initDriver(CIrrDeviceWin32* device)
 	}
 
 #ifdef _IRR_COMPILE_WITH_OPENCL_
-	ocl::COpenCLHandler::getCLDeviceFromGLContext(clDevice,hrc,HDc);
+	ocl::COpenCLHandler::initOpenCL(clDevice,hrc,HDc);
```
Comment (devshgraphicsprogramming) [inline source/Irrlicht/COpenGLDriver.cpp:448]: do not init an OpenCL device context by default for everyone, only if asked for in device creation parameters.

While I'm not asking for a full #162 fix and implementation, that API calls for a query OpenCL devices and a list of devices for which to create contexts in device creation paramters...

Basically a simple `createOpenCLInteropContext` boolean in device creation parameters will do for now.

File: source/Irrlicht/COpenCLHandler.cpp (hunk 1 above)
Comment (manhnt9) [reply source/Irrlicht/COpenCLHandler.cpp:17]: whole COpenCLHandler is current used as static, so everything is static
I will store a COpenCLHandler object in COpenGLDriver, is that OK?

**Comment by devshgraphicsprogramming** [comment]
@manhnt9 any progress on this?

**Comment by manhnt9** [comment]
@devshgraphicsprogramming not yet, I'm developing Saga3D most of the time.

**Comment by devshgraphicsprogramming** [comment]
doing it by myself.

**Pull request closed without merging**